### ISSUE TITLE
Focus system ignores disabled prop in touchables

### ISSUE THREAD

**Issue by pedro-pedrosa** [react-native-community/react-native-tvos#81]
Setting the `disabled` prop on touchables has no effect on the ability to focus the touchable. `onPress` is not triggered when the select button is pressed, as expected.

I wanted to try with RN's `Pressable` component but it doesn't look like we have merged `0.63` yet.

I am testing on android by the way, not sure if it happens on tvos. Decided to report here because it doesn't look like RN core wants to support TV devices anymore.

React Native version:

```
System:
    OS: Linux 5.4 Ubuntu 18.04.5 LTS (Bionic Beaver)
    CPU: (8) x64 Intel(R) Core(TM) i7-4700HQ CPU @ 2.40GHz
    Memory: 242.80 MB / 15.52 GB
    Shell: 4.4.20 - /bin/bash
  Binaries:
    Node: 10.22.0 - /usr/bin/node
    Yarn: 1.21.1 - /usr/bin/yarn
    npm: 6.14.6 - /usr/bin/npm
    Watchman: 4.9.0 - /home/linuxbrew/.linuxbrew/bin/watchman
  SDKs:
    Android SDK: Not Found
  IDEs:
    Android Studio: Not Found
  Languages:
    Java: 11.0.8 - /usr/bin/javac
    Python: 2.7.17 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: ^16.13.1 => 16.13.1 
    react-native: Not Found
    react-native-tvos:  0.62.2-1 
  npmGlobalPackages:
    *react-native*: Not Found
```

## Steps To Reproduce

1. Create an app with several touchables, some of them disabled
2. Cycle through the touchables with directional pad actions (up down left right)

Describe what you expected to happen:
Expected the focus system to skip disabled touchables

Snack, code example, screenshot, or link to a repository:

```tsx
import React from 'react'
import { View, TouchableOpacity, Text } from 'react-native'

const App = () => {
  return (
    <View>
      <TouchableOpacity hasTVPreferredFocus style={{ backgroundColor: '#000'}}>
        <Text style={{ color: '#fff'}}>Button 1</Text>
      </TouchableOpacity>
      <TouchableOpacity disabled style={{ backgroundColor: '#000'}}>
        <Text style={{ color: '#fff'}}>Button 2</Text>
      </TouchableOpacity>
    </View>
  )
}

export default App

```

**Comment by dlowder-salesforce** [comment]
I will have a look at Touchables to see if I can find a common fix for both tvOS and Android TV.  I have not done any specific work yet to make Pressables work on TV.  I'm currently finishing up the first 0.63 release for this repo, and will then have a look at Pressables.

**Comment by dlowder-salesforce** [comment]
Could you try this, and see if it solves the issue on Android TV? It definitely works on Apple TV.

```js
import React from 'react'
import { View, TouchableOpacity, Text } from 'react-native'

const App = () => {
  return (
    <View>
      <TouchableOpacity hasTVPreferredFocus style={{ backgroundColor: '#000'}}>
        <Text style={{ color: '#fff'}}>Button 1</Text>
      </TouchableOpacity>
      <TouchableOpacity disabled isTVSelectable={false} style={{ backgroundColor: '#000'}}>
        <Text style={{ color: '#fff'}}>Button 2</Text>
      </TouchableOpacity>
    </View>
  )
}

export default App
```

**Comment by pedro-pedrosa** [comment]
that sure does work, thanks for this.

did you find this in code? I can't find any documentation about this. typescript says this prop doesn't exist, so if I want to use it I have to resort to module augmentation but I guess that at least I don't have to conditionally render a touchable around my views depending on whether they are disabled which is good news...

**Comment by dlowder-salesforce** [comment]
I think you need to add a separate module to get the Typescript types for TV-specific props.

I'm going to make this automatic by modifying `Libraries/Components/Touchable/TouchableOpacity.js` (and TouchableHighlight) to turn off `isTVSelectable` when `disabled` is true.  I'll add you to the PR.

**Comment by dlowder-salesforce** [comment]
@pedro-pedrosa please take a look at https://github.com/react-native-community/react-native-tvos/commit/3df532a491f1034c053240068c1d0f8566cd7cab and see if that works ok for you.  I've modified one of the examples in RNTester to test that it works.

**Comment by pedro-pedrosa** [comment]
yes this looks great, thank you.

**Comment by pedro-pedrosa** [comment]
sorry I had another look at this and apparently it doesn't work on android. the touchables are still focusable. I even tried to do `focusable={false}` because I saw that in the code from your commit but it didn't work either. so I guess that prop is not implemented in android native code

**Comment by dlowder-salesforce** [comment]
@pedro-pedrosa I'll take a look.... scanning through the touchable and view source files, it looks like `focusable` is the prop that Android TV uses.  I'll try to set up an Android TV test case and verify that.

**Comment by dlowder-salesforce** [comment]
I did some testing, and reviewed the native Android source code. It looks like you want to set the `accessible` prop to false if you want a view to not be focusable on Android.  There are comments in the code that setting the RN prop `focusable` to false does not turn off the native property.

```java
  @ReactProp(name = "focusable")
  public void setFocusable(final ReactViewGroup view, boolean focusable) {
    if (focusable) {
      view.setOnClickListener(
          new View.OnClickListener() {
            @Override
            public void onClick(View v) {
              final EventDispatcher mEventDispatcher =
                  ((ReactContext) view.getContext())
                      .getNativeModule(UIManagerModule.class)
                      .getEventDispatcher();
              mEventDispatcher.dispatchEvent(new ViewGroupClickEvent(view.getId()));
            }
          });

      // Clickable elements are focusable. On API 26, this is taken care by setClickable.
      // Explicitly calling setFocusable here for backward compatibility.
      view.setFocusable(true /*isFocusable*/);
    } else {
      view.setOnClickListener(null);
      view.setClickable(false);
      // Don't set view.setFocusable(false) because we might still want it to be focusable for
      // accessibility reasons
    }
  }
```

**Comment by dlowder-salesforce** [comment]
In the process of investigating this issue, I came across a very interesting post on the UX stackexchange, about why and when disabled controls still need to be accessible to the focus engine: https://ux.stackexchange.com/questions/103239/should-disabled-elements-be-focusable-for-accessibility-purposes

So I'm considering re-doing this so that focusability is fully controlled by the `accessible` prop on all platforms.

**Comment by dlowder-salesforce** [comment]
With commit https://github.com/react-native-community/react-native-tvos/commit/02b89c68f8fd7b7db1e845bec4c08e0746e9f43b , setting `accessible={false}` will remove the touchable from the focus engine on both Android and Apple TV.

**Comment by pedro-pedrosa** [comment]
I can confirm that this works :heavy_check_mark:

**Comment by dlowder-salesforce** [comment]
It will also work on Android if you set `disabled` to true, `accessible` to false.... are you ok with that?

**Comment by dlowder-salesforce** [comment]
This is merged to the `tvos-v0.63.1` release branch, and will be in the next 0.63 npm release.

**Comment by dlowder-salesforce** [comment]
Fixed in 0.63.1-1
